### PR TITLE
moving to ts/react, adding libraries

### PR DESCRIPTION
**Pull request by TheNoeTrevino**
https://github.com/user-attachments/assets/f76c34be-08f3-43e8-ad71-aa627fabc3ea

### CODE REVIEW

File: backend/main/main/settings.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # See https://docs.djangoproject.com/en/5.1/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'django-insecure-6mc+k@gwo)jg*cu*h7=!-yhl@fv11nodonmz5(ir_eowr%k%^h'
```
Comment (TheNoeTrevino) [inline backend/main/main/settings.py:23]: haha...

File: backend/main/music/dynamic_marry.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+from music21 import interval, note, chord
```
Comment (TheNoeTrevino) [inline backend/main/music/dynamic_marry.py:1]: using this as our scale object for generation

File: frontend/src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  { name: "About", path: "/about" },
+];
+const settings = [
+  { name: "Profile", path: "/profile" },
+  { name: "Account", path: "/account" },
+  { name: "Dashboard", path: "/dashboard" },
+  { name: "Logout", path: "/logout" },
+];
+
+function NavBar() {
```
Comment (TheNoeTrevino) [inline frontend/src/components/Navbar.tsx:30]: Honestly, i copied this from here:

https://mui.com/material-ui/react-app-bar/#app-bar-with-responsive-menu

and edited it

File: frontend/src/pages/music-display/SheetMusicDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    setOctaveAnchorEl(event.currentTarget);
+  };
+  const handle16thRhythmClick = (event: MouseEvent<HTMLElement>) => {
+    set16thRhythmAnchorEl(event.currentTarget);
+  };
+  const handle8thRhythmClick = (event: MouseEvent<HTMLElement>) => {
+    set8thRhythmAnchorEl(event.currentTarget);
+  };
+
+  const handleScaleClose = () => {
```
Comment (rockCityMath) [inline frontend/src/pages/music-display/SheetMusicDisplay.tsx:65]: this menu handling stuff seems a little funny, I feel like MUI might have a pre-built way to handle these?

**Comment by rockCityMath** [comment]
I'd have the XML get fetched as soon as they click a menu option instead of having to click "get music"

File: backend/main/music/library.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+from music21 import note, duration
+
+from music21.stream.base import Stream
+
+from .utilities import get_xml_file
+
+sixteenth_variants = ["1111", "112", "121", "211", "0111"]
```
[the remark this answers is not in the document]
Comment (TheNoeTrevino) [reply backend/main/music/library.py:7]: will definitely fix this. I think i am going to come up with a different system entirely to send more complex rhythm objects. I would be more object oriented than just a dictionary

**Pull request merged**